### PR TITLE
QA-14958: Fix saml callback not working if url has parameters

### PR DESCRIPTION
**Pull request by jahia-carp**
https://jira.jahia.org/browse/QA-14958

### CODE REVIEW

File: src/main/java/org/jahia/modules/saml2/actions/SAMLCallback.java [hunk 1]
```diff
@@ -109,7 +109,8 @@ private String retrieveRedirectUrl(HttpServletRequest request, String siteKey) {
             }
         }
 
-        return redirection + "?site=" + siteKey;
+        assert redirection != null;
```
Comment (jkevan) [inline src/main/java/org/jahia/modules/saml2/actions/SAMLCallback.java:112]: What is the purpose of this assertion ?
There is already checks in previous lines, that will anyway set a default `redirection` to "/".
So it's likely impossible for `redirection` to be null at that place.

**Review by jkevan — CHANGES_REQUESTED** [review]

File: src/main/java/org/jahia/modules/saml2/actions/SAMLCallback.java (hunk 1 above)
Comment (jsinovassin) [inline src/main/java/org/jahia/modules/saml2/actions/SAMLCallback.java:112]: Why do we need this assert, redirection will  be at least equals to "/", have a look to the line 108

Comment (jahia-carp) [reply src/main/java/org/jahia/modules/saml2/actions/SAMLCallback.java]: Yes fixed, sorry about that, it was to please my code editor a bit too quickly ;-)

Comment (jahia-carp) [reply src/main/java/org/jahia/modules/saml2/actions/SAMLCallback.java]: thanks, fixed

**Review by jkevan — APPROVED** [review]

**Pull request merged**